lib/rubygems/commands/update_command.rb

```diff
@@ -262,6 +262,11 @@ def update_gems(gems_to_update)
   # Update RubyGems software to the latest version.
 
   def update_rubygems
+    if Gem.disable_system_update_message
+      alert_error Gem.disable_system_update_message
+      return
```

Comment:
I would expect the update attempt to fail (i.e. exit non-zero)

Reply:
Ahh, right. Do you have any preferred exit code for this?

Reply:
fixed @terceiro, it will return 1 now